test/random.js

```diff
@@ -17,200 +19,179 @@ const createSsbServer = SecretStack({
   .use(require('..'))
   .use(require('ssb-gossip'))
 
-let generated = {}
-const F = 100
-const N = 10000
+const AMOUNT_FEEDS = 100
+const AMOUNT_MSGS = 10000
 
-function generateAnimals (ssbServer, feed, f, n, cb) {
-  const a = [feed]
+const CONNECTION_TIMEOUT = 500
+const REPLICATION_TIMEOUT = 2 * CONNECTION_TIMEOUT
 
-  while (f-- > 0) { a.push(ssbServer.createFeed()) }
+function generateAnimals (bot, mainFeed, amountFeeds, amountMsgs, doneCB) {
+  const feeds = [mainFeed]
+  while (amountFeeds-- > 0) {
+    feeds.push(bot.createFeed())
+  }
 
   u.log('generate NAMES')
 
   pull(
-    pull.values(a),
-    paramap(function (feed, cb) {
+    pull.values(feeds),
+    paramap((feed, cb) => {
       const animal = Math.random() > 0.5 ? 'cat' : 'dog'
       const name = animal === 'cat' ? cats.random() : dogs.allRandom()
-
       feed.name = name
-      u.log(feed.id)
+      u.log(feed.id + ' follows itself')
       feed.add(u.follow(feed.id), cb)
     }, 10),
     pull.drain(null, function (err) {
-      if (err) return cb(err)
-      const posts = []
+      if (err) return doneCB(err)
 
+      const posts = []
       pull(
-        pull.count(n),
-        paramap(function (n, cb) {
-          const me = a[~~(Math.random() * a.length)]
+        pull.count(amountMsgs),
+        paramap((i, cb) => {
+          const feed = feeds[~~(Math.random() * feeds.length)]
           const r = Math.random()
-          if (r < 0.1) u.log(n, me.id, r) // log only 1 in 10, less noise
+          // log only 1 in 10, less noise
+          if (r < 0.1) u.log(i, feed.id, r)
           // one in 20 messages is a random follow
```

Comment:
ehh, isn't this ½?

Reply:
Yeah, I mean, I kept some stuff here untouched, so this is Dominic's to answer :laughing:

Reply:
:)